From c101690c546f92104e9a2a9281696fbe934ddfdd Mon Sep 17 00:00:00 2001
From: Meng Dongyang <daniel.meng@rock-chips.com>
Date: Tue, 12 Dec 2017 17:52:36 +0800
Subject: [PATCH] phy: rockchip-inno-usb2: power on phy if host connected when
 resume

In current code, the id irq will be disabled in suspend, if the otg
cable is connected during suspend, the id interrupt will be missing.
This patch checkout the voltage of id pin, power on phy and turn
on vbus if the otg cable is connected.

Change-Id: I988ecdf12e40fdfbc5360896cfb983879ebe23eb
Signed-off-by: Meng Dongyang <daniel.meng@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-usb2.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
index 03ee707d3bbe..1eeffd1ed2bb 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
@@ -1956,6 +1956,20 @@ static int rockchip_usb2phy_pm_resume(struct device *dev)
 					"failed to enable id irq\n");
 				return ret;
 			}
+
+			if (!property_enabled(rphy,
+					      &rport->port_cfg->utmi_iddig)) {
+				dev_dbg(&rport->phy->dev,
+					"port power on when resume\n");
+				extcon_set_state_sync(rphy->edev,
+						      EXTCON_USB_HOST,
+						      true);
+				extcon_set_state_sync(rphy->edev,
+						      EXTCON_USB_VBUS_EN,
+						      true);
+				gpiod_set_value_cansleep(rport->vbus_drv_gpio,
+							 1);
+			}
 		}
 	}
 
-- 
2.35.3

